### ISSUE TITLE
Upgrade to Nokogiri 1.6.x

### ISSUE THREAD

**Issue by mojavelinux** [asciidoctor/asciidoctor#1656]
Upgrade to Nokogiri 1.6.x to coincide with dropping support for Ruby 1.8.

**Comment by mojavelinux** [comment]
Be sure to add the following lines to the Travis CI configuration (.travis.yml) to ensure that the Nokogiri 1.6 gem installs quickly.

```yaml
env:
  global:
    # speed up installation of nokogiri by using system libraries
    - NOKOGIRI_USE_SYSTEM_LIBRARIES=1
```

**Comment by mojavelinux** [comment]
A big problem we're going to run into is that a DocumentFragment.parse bug in Nokogiri >= 1.6.2 breaks tests in JRuby (1.7 or 9k).

**Comment by robertpanzer** [comment]
Oops, does it only break the tests or will it also stop working?
Nokogiri is used for all template based conversion, isn't it?

**Comment by robertpanzer** [comment]
Just seen that Nokogiri 1.6.8.rc1 is packaged with the epub3 backend whilst it is not installed for asciidoctorj(-core) at all.

**Comment by mojavelinux** [comment]
Nokogiri is only used in the test suite, so it only hits us in development. (We only use Nokogiri for Asciidoctor EPUB3 at runtime).

There's a small chance that we're attempting to do something invalid in the test suite with the DocumentFragment parsing which just happens to work in the C version of Nokogiri. We'll need to isolate the exact scenario that is causing the issue in order to determine if this is really a bug or whether we need to make changes to the test code.

Here's the top of the stacktrace:

```
Java::JavaLang::NullPointerException: 
    nokogiri.internals.NokogiriHelpers.renameNode(NokogiriHelpers.java:839)
    nokogiri.XmlNode.relink_namespace(XmlNode.java:499)
    nokogiri.XmlNodeSet.relink_namespace(XmlNodeSet.java:121)
    nokogiri.XmlDocumentFragment.relink_namespace(XmlDocumentFragment.java:208)
    nokogiri.XmlNode.adoptAs(XmlNode.java:1558)
    nokogiri.XmlNode.add_child_node(XmlNode.java:1677)
```

**Comment by robertpanzer** [comment]
Did you see this exception in an asciidoctorJ test?
Because on the current master the asciidoctorj-core tests run without nokogiri.
And asciidoctorj-epub3 is already on sth 1.6.x-ish (1.6.8-rc1)

**Comment by mojavelinux** [comment]
This isn't related to AsciidoctorJ, but rather using JRuby directly. Here's how to test it.

1. Clone the repo
2. Update the version of Nokogiri in asciidoctor.gemspec

3. Use RVM to switch to JRuby, then install the gems
    ```
    $ rvm use jruby@asciidoctor-dev --create
    ```

4. Install the gems
    ```
    $ rm -f Gemfile.lock && bundle
    ```

5. Run the tests
    ```
    rake test
    ```

**Comment by mojavelinux** [comment]
Resolved by #1213.
